src/Middleware/OutputCaching/src/IOutputCacheBufferStore.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.IO.Pipelines;
+
+namespace Microsoft.AspNetCore.OutputCaching;
+
+/// <summary>
+/// Represents a store for cached responses that uses a <see cref="IBufferWriter{byte}"/> as the target.
+/// </summary>
+public interface IOutputCacheBufferStore : IOutputCacheStore
+{
+    /// <summary>
+    /// Gets the cached response for the given key, if it exists.
+    /// If no cached response exists for the given key, <c>null</c> is returned.
+    /// </summary>
+    /// <param name="key">The cache key to look up.</param>
+    /// <param name="destination">The location to which the value should be written.</param>
+    /// <param name="cancellationToken">Indicates that the operation should be cancelled.</param>
+    /// <returns><c>True</c> if the response cache entry if it exists; otherwise <c>False</c>.</returns>
+    ValueTask<bool> TryGetAsync(string key, PipeWriter destination, CancellationToken cancellationToken);
```

Comment:
Am I right in assuming we're using `PipeWriter` instead of `IBufferWriter<byte>` so that responses can be streamed to the caller asynchronously (with backpressure)?

Reply:
Yes, that's the intent - that implementations can respect back-pressure semantics